### DISCUSSION TITLE
New functions

### DISCUSSION THREAD

**Discussion by mr-sugarwolf** [lostlocalhost/METARMap#1] [General]
Has anyone gotten this to work with the new airport format? And, what do the parameters of the new format mean.

**Comment by lostlocalhost** [comment]
> Has anyone gotten this to work with the new airport format? And, what do the parameters of the new format mean.

I haven't touched this in a while, but I updated the code for the new endpoint months ago. I'm not familiar with a new 'airport format' though. Can you share the details?

**Reply by mr-sugarwolf** [reply]
The airport file structure is different from the original format used in Phil Rucker's code. This one has parameters to support the json the document describes. When I ran it I got and error about getting the data. It has been weeks since I tried and did not save the error message.

**Reply by lostlocalhost** [reply]
> The airport file structure is different from the original format used in Phil Rucker's code. This one has parameters to support the json the document describes. When I ran it I got and error about getting the data. It has been weeks since I tried and did not save the error message.

Oh, gotcha! Those were my changes. The 'text' field shows in the top right corner of the led display. The 'display' is a boolean to display that airport or not, and the 'visits' was a field I was going to use to somehow mark my most visited airports, but i never got around to that code.

as for your problem, this code works for me-so i'm not sure what your issue could be.

**Reply by mr-sugarwolf** [reply]
Thank you for the reply and explanation. When I get some time in the next two weeks I will give it another try. I am running this one on a RPi4 just to see if it will work.

**Reply by mr-sugarwolf** [reply]
I finally got to continue with this today. Here is the error I am getting. Any suggestion are appreciated. Thank you.
Traceback (most recent call last):
  File "/home/pi/metar.py", line 135, in <module>
    airport_dict = json.loads(data)
                   ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 89 column 20 (char 1548)

**Reply by lostlocalhost** [reply]
there's something wrong with your airports file that's failing to load. verify your json is correctly formatted.

**Reply by mr-sugarwolf** [reply]
Thank you. I thought so, too. After hours of looking at it I found nothing. I took a break and came back to it looking line by line and character by character. I finally found a missing end quote in a  text param for one airport.

This was reminiscent of a BASIC program I was running almost 40 years ago. The program has been running for months without a problem. I had to add update some data. It ran up to one data point and would stop. After hours of looking at it and not finding the problem, I asked a co-worker to take a look. In less than 2 minutes he said, "You are missing a comma right here." duh

Working on final error message. I made sure active display is set to True. Line 275 is displaymetar..none(disp). I cannot find the 'NoneType' attribute.
pi@METARPi4:~ $ sudo python3 metar.py
Traceback (most recent call last):
  File "/home/pi/metar.py", line 275, in <module>
    displaymetar.none(disp)
    ^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'none'

Found my displaymetar.py was missing. Now, have to get the info to show on the display. Wiring checked. Packages checked and updated. And airports file checked. No errors shown when the script runs.

**Reply by lostlocalhost** [reply]
Saying it's all good now?

That line is for calling this function, but looks like you figured out your issue.
https://github.com/lostlocalhost/METARMap/blob/19ee84b7c301f152f1ceee21ec96ef6a346b011a/displaymetar.py#L110

**Reply by mr-sugarwolf** [reply]
Still working on getting the display to work. It is saying it failed to get size when I run it.
Traceback (most recent call last):
  File "/home/pi/metar.py", line 267, in <module>
    displaymetar.outputMetar1(disp, displayList[displayAirportCounter], conditionDict.get(displayList[displayAirportCounter][0], None))
  File "/home/pi/displaymetar.py", line 66, in outputMetar1
    w,h = fontXSmall.getsize(str(station[1])) 
          ^^^^^^^^^^^^^^^^^^
AttributeError: 'FreeTypeFont' object has no attribute 'getsize' The longer lines are cut off, of course.

This is the line in displaymetar.py (#66) that is referenced.
63         #draw.line([(x + 85, top + 18), (x + 85, bottom)], fill=255, width=>
 64         central = timezone('US/Central')
 65         draw1.text((x, top + 0), station[0][1:] + "-" + condition["flightCa>
 66         w,h = fontXSmall.getsize(str(station[1]))

**Reply by lostlocalhost** [reply]
I just made an update that may be related to what you were seeing. When the data source returns an empty flightCategory, that caused an error on the display because your can't concatenate a None object.
It now checks for this.
When flightCategory is not present, it will light up white and display as UNK in the display.

**Reply by mr-sugarwolf** [reply]
Thank you. I will give it a try next week. I am tied up this week.

**Comment by mr-sugarwolf** [comment]
With the exception of getting information on the display, yes. It may be the display itself. I am going to get another one. I compared the lights being lit when I run it to the one I have had running with Phil’s code for the past three years.Thank you for your help. I really appreciate it. Looking forward to see the difference between the displayed information.
    Julius
  

 From: Brian Smith ***@***.***>Sent: Monday, April 1, 2024 9:13 AMTo: lostlocalhost/METARMap ***@***.***>Cc: mr-sugarwolf ***@***.***>; Author ***@***.***>Subject: Re: [lostlocalhost/METARMap] New functions (Discussion #1) 
Saying it's all good now?
That line is for calling this function, but looks like you figured out your issue.
https://github.com/lostlocalhost/METARMap/blob/19ee84b7c301f152f1ceee21ec96ef6a346b011a/displaymetar.py#L110

**Reply by mr-sugarwolf** [reply]
With the exception of getting information on the display, yes. It may be the display itself. I am going to get another one. I compared the lights being lit when I run it to the one I have had running with Phil’s code for the past three years.

Thank you for your help. I really appreciate it. Looking forward to see the difference between the displayed information.
